cmd/flarectl/firewall.go

```diff
@@ -0,0 +1,380 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/codegangsta/cli"
+)
+
+func firewallAccessRules(c *cli.Context) {
+	if err := checkEnv(); err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	organizationID, zoneID, err := getScope(c)
+	if err != nil {
+		return
+	}
+
+	// Create an empty access rule for searching for rules
+	rule := cloudflare.AccessRule{
+		Configuration: getConfiguration(c),
+	}
+	if c.String("scope-type") != "" {
+		rule.Scope.Type = c.String("scope-type")
+	}
+	if c.String("notes") != "" {
+		rule.Notes = c.String("notes")
+	}
+	if c.String("mode") != "" {
+		rule.Mode = c.String("mode")
+	}
+
+	var rules []cloudflare.AccessRule
+	if organizationID != "" {
```

Comment:
Arguably a bare `switch` statement is more idiomatic here. It also appears that a large chunk of logic is duplicated between all three cases, this could (probably?) be deduped.

Reply:
I haven't found a good way to deduplicate the logic, because the calls to key functions vary both in name and number of arguments. I'm open to suggestions, though. I can use a bare switch no problem. I was trying to be consistent with the use of if/else if/else over bare switch from zone.go

Reply:
Sounds good; consistency is probably more important and I'm not familiar with this code base. Thanks!